agreement/actions.go

```diff
@@ -232,19 +232,21 @@ func (a ensureAction) do(ctx context.Context, s *Service) {
 		logEvent.Type = logspec.RoundConcluded
 		s.log.with(logEvent).Infof("committed round %d with pre-validated block %v", a.Certificate.Round, a.Certificate.Proposal)
 		s.log.EventWithDetails(telemetryspec.Agreement, telemetryspec.BlockAcceptedEvent, telemetryspec.BlockAcceptedEventDetails{
-			Address: a.Certificate.Proposal.OriginalProposer.String(),
-			Hash:    a.Certificate.Proposal.BlockDigest.String(),
-			Round:   uint64(a.Certificate.Round),
+			Address:     a.Certificate.Proposal.OriginalProposer.String(),
+			Hash:        a.Certificate.Proposal.BlockDigest.String(),
+			Round:       uint64(a.Certificate.Round),
+			ValidatedAt: a.Payload.validatedAt,
```

Comment:
Could you add to the `telemetryspec.BlockAcceptedEventDetails` another field to indicate whether it's a prevalidated block or not ?
( i.e. so that this branching would be visible on the telemetry ? )